src/main/java/com/redhat/devtools/lsp4ij/LanguageServerWrapper.java

```diff
@@ -630,43 +631,16 @@ public boolean canOperate(Project project) {
      *     the method return a CompletableFuture which returns null.
      * </p>
      *
-     * @param file     the file to connect to the language server
-     * @param document the document of the file and null otherwise. In the null case, the document will be retrieved from the file
-     *                 by using a blocking read action.
-     * @return the completable future with the language server instance or null.
-     */
-    CompletableFuture<@Nullable LanguageServer> connect(@NotNull VirtualFile file,
-                                                        @Nullable Document document) {
-        if (file != null && file.exists()) {
-            return connect(LSPIJUtils.toUri(file), file, document);
-        }
-        return CompletableFuture.completedFuture(null);
-    }
-
-    /**
-     * Connect the given file Uri to the language server and returns the language server instance when:
-     *
-     * <ul>
-     *     <li>language server is initialized</li>
-     *     <li>didOpen for the given file Uri happens</li>
-     * </ul>
-     *
-     * <p>
-     *     In other case (ex : language server initialize future is null, file cannot be retrieved by the given Uri),
-     *     the method return a CompletableFuture which returns null.
-     * </p>
-     *
-     * @param fileUri          the file Uri to connect to the language server
-     * @param optionalFile     the file which matches the non-nullable file uri and null otherwise. In the null case, the file will be retrieved by the file Uri.
+     * @param file             the file to connect to the language server
      * @param optionalDocument the document of the file and null otherwise. In the null case, the document will be retrieved from the file
      *                         by using a blocking read action.
      * @return the completable future with the language server instance or null.
      */
-    private CompletableFuture<LanguageServer> connect(@NotNull URI fileUri,
-                                                      @Nullable VirtualFile optionalFile,
-                                                      @Nullable Document optionalDocument) {
+    CompletableFuture<@Nullable LanguageServer> connect(@Nullable VirtualFile file,
+                                                        @Nullable Document optionalDocument) {
         removeStopTimer(false);
 
+        URI fileUri = toUri(file);
```

Comment:
This is a potential NPE since `file` is `@Nullable` and `toUri()` requires a non-null `VirtualFile` as its parameter.

Reply:
Good catch! fixed